### PR TITLE
applying accepted v3 filesEditOptions enhancements to v2

### PR DESCRIPTION
**Pull request by ATorrise**
**What It Does**
Copying changes from [v3](https://github.com/zowe/zowe-cli/pull/2027) to v2

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: packages/imperative/src/imperative/src/plugins/utilities/npm-interface/uninstall.ts
Comment (ATorrise) [inline packages/imperative/src/imperative/src/plugins/utilities/npm-interface/uninstall.ts:1]: when building, this file came up in linter. can remove from this PR if you guys would like

File: packages/cli/__tests__/zosfiles/__unit__/edit/Edit.utils.unit.test.ts [hunk 1]
```diff
@@ -206,41 +208,7 @@ describe("Files Edit Utilities", () => {
             });
         });
     });
-    describe("localDownload()", () => {
```
Comment (awharn) [inline packages/cli/__tests__/zosfiles/__unit__/edit/Edit.utils.unit.test.ts:209]: Should this test suite be removed? It seems to be handling tests for downloading files from the mainframe.

Comment (t1m0thyj) [reply packages/cli/__tests__/zosfiles/__unit__/edit/Edit.utils.unit.test.ts:209]: Agree that these tests probably shouldn't be removed.

The [v3 PR to add edit options](https://github.com/zowe/zowe-cli/pull/2027) added 8 tests and removed 0.

I would expect this PR to do the same, but currently it adds 4 tests and removes 2.

Comment (ATorrise) [reply packages/cli/__tests__/zosfiles/__unit__/edit/Edit.utils.unit.test.ts]: tests are present, view file to and search for '(did not write to temp location)'

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @ATorrise for applying this enhancement to v2 and updating the description of `--editor` option for consistency!

**Review by awharn — APPROVED** [review]

**Review by gejohnston — APPROVED** [review]

**Pull request merged**